tanner/emulators/rfi.py

```diff
@@ -56,20 +52,20 @@ async def download_file(self, path):
                     rfile.write(data.encode('utf-8'))
         return file_name
 
-    def download_file_ftp(self, url):
+    async def download_file_ftp(self, url):
         host = url.host
-        ftp_path = url.path.rsplit('/', 1)[0][1:]
+        ftp_path = url.path
         name = url.name
         try:
-            ftp = ftplib.FTP(host)
-            ftp.login()
-            ftp.cwd(ftp_path)
+            ftp = aioftp.Client()
```

Comment:
you can use `async with` context for session, so you don't need to login and quit explicitly

Reply:
ok ill give it try